static/js/sandbox-main.js

```diff
@@ -48,3 +48,138 @@ mobileNavTriggerButton.addEventListener('click', function () {
     mobileNavIcon.innerHTML = feather.icons['menu'].toSvg();
   }
 });
+
+// Sanbox Dashboard UI Application List Interactions
+// -------------------------------------------------
+
+// Dashboard App Credential Toggle
+// This shows/hides the client ID and secret for each application
+const appCredentialToggleLinks = document.querySelectorAll('.app-credentials-toggle');
+
+// Toggle on Mouse Click
+appCredentialToggleLinks.forEach((appCredentialToggleLink) => {
+  appCredentialToggleLink.addEventListener('click', function() {
+    let appID = document.querySelector('#id-' + this.id);
+    let appSecret = document.querySelector('#secret-' + this.id);
+    if (appID.type == 'password') {
+      appID.type = 'text';
+      appSecret.type = 'text';
+    } else {
+      appID.type = 'password';
+      appSecret.type = 'password';
+    }
+  });
+});
+
+// Toggle for Keyboard Naviagation
+appCredentialToggleLinks.forEach((appCredentialToggleLink) => {
+  appCredentialToggleLink.addEventListener('keypress', function (e) {
+    var key = e.which || e.keyCode;
+    if (key === 13) { // 13 is enter
+      let appID = document.querySelector('#id-' + this.id);
+      let appSecret = document.querySelector('#secret-' + this.id);
+      if (appID.type == 'password') {
+        appID.type = 'text';
+        appSecret.type = 'text';
+      } else {
+        appID.type = 'password';
+        appSecret.type = 'password';
+      }
+    }
+  });
+});
+
+// Copy Credentials to Clipboard
+function copyCredential(copyID) {
+  // Determine if a user is copying a client id or secret
+  if (copyID.includes('secret') === true) {
+    // Get the element we want to copy
+    let secretToCopy = document.querySelector('#' + copyID);
+    // If it is hidden, change to text, copy, and then hide again
+    if (secretToCopy.type == 'password') {
+      secretToCopy.type = 'text';
+      secretToCopy.select();
+      document.execCommand('copy');
+      secretToCopy.type = 'password';
+    } else {
+      // Else, if it is shown, just copy!
+      secretToCopy.select();
+      document.execCommand('copy');
+    }
+    // Fade In / Out Copy Confirmation Message
+    let copyConfirmation = document.querySelector('#confirm-' + copyID);
+    $( copyConfirmation ).fadeIn( 300 ).delay( 1500 ).fadeOut( 400 );
+  } else {
```

Comment:
I'm not seeing a difference between the logic in this block and the above block.

Reply:
I guess it is kind of redundant - and the underlying logic is the same, but I did it like this to make it easy to target the `id` verses `secret`. If there's a cleaner way to do that - I am open to work on this a little bit more.

Reply:
it doesn't look like the code behaves differently if it's acting on an `id` or a `secret`. Can the conditional `copyID.includes('secret') === true` be removed?

Reply:
As discussed, you were totally right. This was redundant. Consolidated everything down to one function and committed the fix.